File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/common/RJSTrackLocation.java
package fr.sncf.osrd.railjson.schema.common;

import com.squareup.moshi.Json;
import fr.sncf.osrd.railjson.schema.infra.RJSTrackSection;

public final class RJSTrackLocation {
    @Json(name = "track_section")
    public ID<RJSTrackSection> trackSection;

    public double offset;
}

File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/common/RJSTrackLocation.kt
package fr.sncf.osrd.railjson.schema.common

import com.squareup.moshi.Json
import fr.sncf.osrd.railjson.schema.infra.RJSTrackSection

class RJSTrackLocation(
    @Json(name = "track_section") var trackSection: ID<RJSTrackSection>,
    var offset: Double,
)
